src/Window.cpp

```diff
@@ -170,6 +170,19 @@ wlr_box CWindow::getWindowInputBox() {
     return finalBox;
 }
 
+wlr_box CWindow::getWindowGroupBarBox() {
+    const auto PWORKSPACE      = g_pCompositor->getWorkspaceByID(m_iWorkspaceID);
```

Comment:
wouldn't this be better as a generic?

getWindowDecorationBox(type)

Reply:
you could make another func then `isPointInDecoration` with an additional `bool` parameter of `ignoreWindow`

Reply:
I would have to either pass a decoration type (wouldn't allow duplicated decorations) or the pointer to the decoration (would create redundant code)

maybe a solution is adding functions for generalized decorations and basing the groupbar functions on those (which doesn't remove the need for the function)

Reply:
why would we allow duplicate decorations?

Reply:
idk, for now I don't believe we want duplicate decos. `getDecorationByType` and `removeDecorationByType` already exist

Reply:
seems like decorations don't depend on a window
should each decoration implement it's own `getDecorationBox()` function?
I was trying to put that in the class `IHyprWindowDecoration`

Reply:
> seems like decorations don't depend on a window

they do...?

> should each decoration implement it's own getDecorationBox() function?

probably

Reply:
the m_pWindow isn't implemented in `IHyprWindowDecoration`, but in each of the decorations
that would have to be moved to `IHyprWindowDecoration` to implement a generalized `getDecorationBox()` function
I think for now I'm just gonna put it in `CHyprGroupBarDecoration`

Reply:
`m_pWindow` can be moved to the deco iface

Reply:
generally looks good but the switch in the box get func is an odd method for sure

Reply:
need to separate when when decoration is just in one side of the window, since the box won't include the window itself
no idea how to do that easier with bool algebra

should I add in PR for review?

Reply:
you can commit yea, easier to read.

Reply:
done